packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -14,6 +14,73 @@ import 'gesture_detector.dart';
 import 'navigator.dart';
 import 'transitions.dart';
 
+class SemanticsClipper extends SingleChildRenderObjectWidget{
+  const SemanticsClipper({
+    super.key,
+    super.child,
+    this.clipDetailsNotifier,
+  });
+
+  final ValueNotifier<EdgeInsets>? clipDetailsNotifier;
+
+  @override
+  RenderSemanticsClipper createRenderObject(BuildContext context) {
+    return RenderSemanticsClipper(clipDetailsNotifier: clipDetailsNotifier,);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderSemanticsClipper renderObject) {
+    renderObject.clipDetailsNotifier = clipDetailsNotifier;
+  }
+}
+
+class RenderSemanticsClipper extends RenderProxyBox {
+  RenderSemanticsClipper({
+    // required ClipDirection clipDirection,
+    ValueNotifier<EdgeInsets>? clipDetailsNotifier,
```

Comment:
should this be non nullable?

Reply:
I'm thinking of if the notifier is null the clipper does not listen to anything and always uses EdgeInsets.zero to clip the semantics rect, which preserves its original size. If you don't think that is necessary we can make it non-nullable